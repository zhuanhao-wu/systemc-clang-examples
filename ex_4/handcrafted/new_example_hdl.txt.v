module fa_0(
input wire [0:0] a,
input wire [0:0] b,
input wire [0:0] cin,
output reg [0:0] cout,
output reg [0:0] s 
);



endmodule // fa_0
module fa_1(
input wire [0:0] a,
input wire [0:0] b,
input wire [0:0] cin,
output reg [0:0] cout,
output reg [0:0] s 
);



endmodule // fa_1
module fa_2(
input wire [0:0] a,
input wire [0:0] b,
input wire [0:0] cin,
output reg [0:0] cout,
output reg [0:0] s 
);



endmodule // fa_2
module fa_3(
input wire [0:0] a,
input wire [0:0] b,
input wire [0:0] cin,
output reg [0:0] cout,
output reg [0:0] s 
);



endmodule // fa_3