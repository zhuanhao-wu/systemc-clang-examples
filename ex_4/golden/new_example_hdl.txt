  [
    (hModule fa_0)
    [
      (hPortin a)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin b)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin cin)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortout cout)
      [
        (hType sc_out)
        (hType _Bool)
      ]
      (hPortout s)
      [
        (hType sc_out)
        (hType _Bool)
      ]
    ]
    [
    ]
  ]
  [
    (hModule fa_1)
    [
      (hPortin a)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin b)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin cin)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortout cout)
      [
        (hType sc_out)
        (hType _Bool)
      ]
      (hPortout s)
      [
        (hType sc_out)
        (hType _Bool)
      ]
    ]
    [
    ]
  ]
  [
    (hModule fa_2)
    [
      (hPortin a)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin b)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin cin)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortout cout)
      [
        (hType sc_out)
        (hType _Bool)
      ]
      (hPortout s)
      [
        (hType sc_out)
        (hType _Bool)
      ]
    ]
    [
    ]
  ]
  [
    (hModule fa_3)
    [
      (hPortin a)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin b)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortin cin)
      [
        (hType sc_in)
        (hType _Bool)
      ]
      (hPortout cout)
      [
        (hType sc_out)
        (hType _Bool)
      ]
      (hPortout s)
      [
        (hType sc_out)
        (hType _Bool)
      ]
    ]
    [
    ]
  ]
